packages/syft/src/syft/grid/client/client.py

```diff
@@ -151,13 +155,36 @@ def login(
         credentials = {"email": email, "password": password}
 
     # connecting to domain
-    try:
-        node = connect(
-            url=grid_url, credentials=credentials, conn_type=conn_type, timeout=timeout
-        )
-    except requests.ConnectTimeout:
-        raise requests.ConnectTimeout(
-            f"Connection to node with: {url} timed out. Please try again !!!"
+    node = None
+    timeout_btw_retries = timeout // retry
+    retry_attempt = 1
+
+    while node is None and retry_attempt < retry:
+        try:
+            node = connect(
+                url=grid_url,
+                credentials=credentials,
+                conn_type=conn_type,
+                timeout=timeout,
+            )
+        except requests.ConnectTimeout:
+            raise requests.ConnectTimeout(
+                f"Connection to node with: {url} timed out. Please try again !!!"
+            )
+        except requests.ConnectionError as e:
+            if retry_attempt < retry:
```

Comment:
Should it be <= , so we try exactly the retry times.

Reply:
Fixed in: https://github.com/OpenMined/PySyft/pull/6578/commits/56b1d0d30d0b66964a8ecfd1be57d47a4d61d2de